### DISCUSSION TITLE
How to run Target in backgroud before debugging

### DISCUSSION THREAD

**Discussion by Eptagone** [dotnet/msbuild#8217] [Q&A]
I want to run a Node Script command in background before my debugging process.
```xml
<Target Name="DebugRunAssetsDevServer" BeforeTargets="Run" Condition=" '$(Configuration)' == 'Debug' ">
    <!-- Run the assets dev server -->
    <Message Importance="normal" Text="Starting Assets Dev Server" />
    <Exec Command="npm run dev" ContinueOnError="true" />
</Target>
```
The above code doesn't work when I start the project in debug mode. It works if I change to `BeforeTargets="Build"` but hangs waiting for the command to finish. But that command shouldn't finish because it starts my Vite Dev Server which compiles my scripts while I'm editing them.

**Comment by JaynieBai** [comment]
https://learn.microsoft.com/en-us/visualstudio/msbuild/target-build-order?view=vs-2022. Seems msbuild targets can only be build in order. I think you need to build your DebugRunAssetsDevServer target seperately.

**Comment by rainersigwald** [comment]
You _can_ start a long-running process using shell operations (`Exec` runs its argument in a `cmd.exe` or `/bin/sh` shell, so you can use any of the normal break-away mechanisms), but you often don't want to do this, because it would make the "run a build,, edit something, run a build" behavior very confusing--starting and never stopping two dev servers.
